android/app/src/main/java/com/sphereon/ssi/wallet/MainActivity.java

```diff
@@ -71,5 +123,54 @@ public void invokeDefaultOnBackPressed() {
     super.invokeDefaultOnBackPressed();
   }
 
+  @Override
+  protected void onStop() {
+    super.onStop();
+    if (isAppMovingToBackground()) {
+      backgroundHandler.postDelayed(backgroundRunnable, Constants.BACKGROUND_DELAY);
+    }
+  }
+
+  private boolean isAppMovingToBackground() {
+    ActivityManager activityManager = (ActivityManager) getSystemService(Context.ACTIVITY_SERVICE);
+    if (activityManager == null) {
+      return false;
+    }
+
+    List<ActivityManager.RunningTaskInfo> tasks = activityManager.getRunningTasks(1);
```

Comment:
See https://developer.android.com/reference/android/app/ActivityManager#getRunningTasks(int). 

It is deprecated as of 21 and will be removed (or is even removed in newer versions)
It is also only meant for debug purposes or task applications according to the link

Reply:
this has completely been refactored